### PR TITLE
chore(common): loosen .keyman-touch-layout schema layer id requirements

### LINKED ISSUE #9843: bug(common): keyman-touch-layout schema has restrictive id pattern that some keyboards do not match
I was trying to find usage of an underscore layer ID in a keyman-touch-layouts file, when I saw 
[gff_amharic](https://raw.githubusercontent.com/keymanapp/keyboards/master/release/gff/gff_amharic/source/gff_amharic.keyman-touch-layout) had some non ascii characters in the layer ID
e.g. "id": "ዐ-layer",

_Originally posted by @darcywong00 in https://github.com/keymanapp/api.keyman.com/pull/210#discussion_r1369741485_

### PR DESCRIPTION
**Pull request by mcdurdin**
Fixes #9843.

See also keymanapp/api.keyman.com#243

@keymanapp-test-bot skip

### CODE REVIEW

File: common/schemas/keyman-touch-layout/README.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
 # keyman-touch-layout
 
-Documentation at https://help.keyman.com/developer/...TODO...
+Documentation at https://github.com/keymanapp/keyman/tree/master/common/schemas/keyman-touch-layout/README.md
 
 * Primary version:
   * https://github.com/keymanapp/keyman/tree/master/common/schemas/keyman-touch-layout
 * Deployed to:
-  * N/A
+  * https://github.com/keymanapp/api.keyman.com/tree/master/schemas/keyman-touch-layout
```
Comment (darcywong00) [inline common/schemas/keyman-touch-layout/README.md:8]: I saw the clean spec deployed to
https://github.com/keymanapp/api.keyman.com/tree/master/schemas/keyman-touch-layout/2.1.1

Where does that come from?

Comment (mcdurdin) [reply common/schemas/keyman-touch-layout/README.md:8]: > Where does that come from?

The clean spec is also in the keyman repo.

**Review by darcywong00 — APPROVED** [review]
lgtm

**Comment by keyman-server** [comment]
Changes in this pull request will be available for download in [Keyman version 17.0.280-beta](https://keyman.com/downloads/releases/beta/17.0.280)

**Pull request merged**